diff --git a/test/test_basic.py b/test/test_basic.py
index 81ec47c..94bc1da 100644
--- a/test/test_basic.py
+++ b/test/test_basic.py
@@ -101,10 +101,3 @@ class TestBasic(unittest.TestCase):
         subprocess.check_call(['scons', 'py', '--prefix=' + prefix],
                               cwd=self.tgt.parent.parent,
                               shell=True)
-
-    def test_boost_python_pattern_generator(self):
-        """Test `boost_python_pattern` generator"""
-        plugin = autowig.generator.plugin
-        autowig.generator.plugin = 'boost_python_pattern'
-        self.test_mapping_export()
-        autowig.generator.plugin = plugin
