diff --git a/test/test_basic.py b/test/test_basic.py
index cfa2781..f277b49 100644
--- a/test/test_basic.py
+++ b/test/test_basic.py
@@ -102,13 +102,6 @@ class TestBasic(unittest.TestCase):
                               cwd=self.tgt.parent.parent,
                               shell=True)
 
-    def test_pyclanglite_parser(self):
-        """Test `pyclanglite` parser"""
-        plugin = autowig.parser.plugin
-        autowig.parser.plugin = 'pyclanglite'
-        self.test_mapping_export()
-        autowig.parser.plugin = plugin
-
     def test_boost_python_pattern_generator(self):
         """Test `boost_python_pattern` generator"""
         plugin = autowig.generator.plugin
